src/toolchains.ts

```diff
@@ -0,0 +1,133 @@
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+import { create as xmlCreate } from 'xmlbuilder2';
+import * as constants from './constants';
+import { getBooleanInput } from './util';
+
+export async function configureToolchains(jdkHome: string) {
+  const version = core.getInput(constants.INPUT_JAVA_VERSION);
+  const vendor = core.getInput(constants.INPUT_DISTRIBUTION);
```

Comment:
We already read these in the `run()` function (in `setup-java.ts`).  Can you just pass them as arguments to this function?

Reply:
Fixed